src/register/macros.rs

```diff
@@ -1,13 +1,13 @@
 macro_rules! read_csr {
-    ($csr_number:expr, $asm_fn: ident) => {
+    ($csr_number:literal, $asm_fn: ident) => {
```

Comment:
The change from `expr` to `literal` is to prevent expressions like `1<<2` (it will be stringified as "1<<2") from being accidentally accepted.